src/stress_stream/producer.py

```diff
@@ -0,0 +1,183 @@
+import asyncio
+import os
+import random
+import struct
+from datetime import datetime
+
+import typer
+from prometheus_client import Counter, Histogram
+from streamstore import S2, Stream
+from streamstore.schemas import AppendInput, Record
+
+from stress_stream._constants import HISTOGRAM_BUCKETS, METRICS_NAMESPACE, S2_AUTH_TOKEN
+from stress_stream._utils import metrics_server
+
+CYCLE_EVERY = 0.5  # 0.5s or 500ms
+MAX_JITTER = 0.2  # 0.2s or 200ms
+TIMESTAMP_HEADER_NUM_BYTES = 10
+
+
+appended_records_counter = Counter(
+    name="appended_records_total",
+    documentation="Total number of records successfully appended.",
+    namespace=METRICS_NAMESPACE,
+)
+appended_bytes_counter = Counter(
+    name="appended_bytes_total",
+    documentation="Total number of bytes successfully appended.",
+    namespace=METRICS_NAMESPACE,
+)
+append_latency_histogram = Histogram(
+    name="append_latency_seconds",
+    documentation="Time taken for each append operation in seconds.",
+    namespace=METRICS_NAMESPACE,
+    buckets=HISTOGRAM_BUCKETS,
+)
+append_errors_counter = Counter(
```

Comment:
Can you also add a counter for total append operations, then becomes easy to compute success or failure rate